Configure

```diff
@@ -1110,8 +1110,8 @@ $target{exe_extension}=".exe" if ($config{target} eq "DJGPP"
 $target{exe_extension}=".pm"  if ($config{target} =~ /vos/);
 
 ($target{shared_extension_simple}=$target{shared_extension})
-    =~ s|\.\$\(SHLIB_VERSION_NUMBER\)||;
-$target{dso_extension}=$target{shared_extension_simple};
+    =~ s|\.\$\(SHLIB_VERSION_NUMBER\)|| unless $target{shared_extension_simple};
+$target{dso_extension}=$target{shared_extension_simple} unless $target{dso_extension};
```

Comment:
Another way to write this would be:

``` perl
($target{shared_extension_simple}||=$target{shared_extension})
    =~ s|\.\$\(SHLIB_VERSION_NUMBER\)||;
$target{dso_extension}||=$target{shared_extension_simple};
```

Reply:
Cool! Though I've decided to go for defined($target{x}). This would allow to use empty string for extension. If so required for whatever reason. Then I reckoned that replacing shared_extension_simple assignment without `unless` is not actually right. Because it would prevent user from specifying for example `.$(SHLIB_VERSION_NUMBER).ext`, which would be forced to `.ext`. I mean if shared_extension_simple is defined, then it shouldn't be processed with `=~`. Squashed and pushed.